google/cloud/cloudbuild/cloud_build_connection.cc

```diff
@@ -0,0 +1,679 @@
+}
+
+StreamRange<google::devtools::cloudbuild::v1::WorkerPool>
+CloudBuildConnection::ListWorkerPools(
+    google::devtools::cloudbuild::v1::ListWorkerPoolsRequest request) {
+  return google::cloud::internal::MakePaginationRange<
+      StreamRange<google::devtools::cloudbuild::v1::WorkerPool>>(
+      std::move(request),
+      [](google::devtools::cloudbuild::v1::ListWorkerPoolsRequest const&) {
+        return StatusOr<
+            google::devtools::cloudbuild::v1::ListWorkerPoolsResponse>{};
+      },
+      [](google::devtools::cloudbuild::v1::ListWorkerPoolsResponse const&) {
+        return std::vector<google::devtools::cloudbuild::v1::WorkerPool>();
+      });
+}
+
+namespace {
+class CloudBuildConnectionImpl : public CloudBuildConnection {
+ public:
+  CloudBuildConnectionImpl(
+      std::unique_ptr<google::cloud::BackgroundThreads> background,
+      std::shared_ptr<cloudbuild_internal::CloudBuildStub> stub,
+      Options const& options)
+      : background_(std::move(background)),
+        stub_(std::move(stub)),
+        retry_policy_prototype_(
+            options.get<CloudBuildRetryPolicyOption>()->clone()),
+        backoff_policy_prototype_(
+            options.get<CloudBuildBackoffPolicyOption>()->clone()),
+        polling_policy_prototype_(
+            options.get<CloudBuildPollingPolicyOption>()->clone()),
+        idempotency_policy_(
+            options.get<CloudBuildConnectionIdempotencyPolicyOption>()
+                ->clone()) {}
+
+  ~CloudBuildConnectionImpl() override = default;
+
+  future<StatusOr<google::devtools::cloudbuild::v1::Build>> CreateBuild(
+      google::devtools::cloudbuild::v1::CreateBuildRequest const& request)
+      override {
+    auto stub = stub_;
+    return google::cloud::internal::AsyncLongRunningOperation<
+        google::devtools::cloudbuild::v1::Build>(
+        background_->cq(), request,
+        [stub](google::cloud::CompletionQueue& cq,
+               std::unique_ptr<grpc::ClientContext> context,
+               google::devtools::cloudbuild::v1::CreateBuildRequest const&
+                   request) {
+          return stub->AsyncCreateBuild(cq, std::move(context), request);
+        },
+        [stub](google::cloud::CompletionQueue& cq,
+               std::unique_ptr<grpc::ClientContext> context,
+               google::longrunning::GetOperationRequest const& request) {
+          return stub->AsyncGetOperation(cq, std::move(context), request);
+        },
+        [stub](google::cloud::CompletionQueue& cq,
+               std::unique_ptr<grpc::ClientContext> context,
+               google::longrunning::CancelOperationRequest const& request) {
+          return stub->AsyncCancelOperation(cq, std::move(context), request);
+        },
+        &google::cloud::internal::ExtractLongRunningResultResponse<
+            google::devtools::cloudbuild::v1::Build>,
+        retry_policy_prototype_->clone(), backoff_policy_prototype_->clone(),
```

Comment:
Please rebase and regenerate to pick up the generator changes from #8013

Reply:
Done.

Reply:
Thanks